File: JAVA_CODING/JAVA_DSA/LL.java
public class LL {

    class Node{
        String data;
        Node next;
    
        Node(String data) {
            this.data = data;
            this.next = null;
        }
    }
    Node head;
    public void addFirst(String data){
        Node newNode  = new Node(data);
        if(head ==null){
            head = newNode;
            return;
        }
        newNode.next = head;
        head = newNode;
    }
    public void addLast(String data){
        Node newNode  = new Node(data);
        if(head ==null){
            head = newNode;
            return;
        }
        Node currNode = head;
        while(currNode.next != null){
            currNode = currNode.next;
        }
    }


    public static void main(String[] args) {
        LL list = new LL();
        list.addFirst("a");
        list.addFirst("b");
        list.addFirst("c");
        
    }
}

File: CPP_CODING/LL.cpp
#include <iostream>
using namespace std;

class Node
{
public:
    int data;
    Node *next;

    Node(int val)
    {
        data = val;
        next = NULL;
    }
};

void insertAtLast(Node *&head, int val)
{
    Node *n = new Node(val);
    if (head == NULL)
    {
        head = n;
        return;
    }

    Node *temp = head;
    while (temp->next != NULL){
        temp = temp->next;
    }
    temp->next = n;
}

void printLL(Node* LL){
    while(LL != NULL){
        printf("%d \n", LL->data);
        LL = LL->next;
    }
}


int main(){
    Node* head = nullptr;

    insertAtLast(head, 5);
    insertAtLast(head, 6);
    printf("printing the list: \n");
    printLL(head);
    getchar();
}